tests/test_injection.py

```diff
@@ -116,11 +116,13 @@ def test_constructor(self):
                 == provider.require_response
             )
 
-        # Asserting that we are not leaking providers references
+        # Asserting that some providers could still have access to some parts
+        # of the injector since the injector is passed when instantiating the
+        # provider.
         weak_ref = weakref.ref(injector.providers[0])
         assert weak_ref()
         del injector
-        assert weak_ref() is None
+        assert weak_ref()
```

Comment:
I'm not sure what's the purpose of this test. Previously, it was checking that objects are deallocated properly. Now it's checking that they are not? :)

Reply:
You're right, this test has lost its usefulness when we started to provide the `Injector` directly instead of the `Crawler`. I've removed this test altogether.